### PR TITLE
Default Detail Comment Fix

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5203

*Description of changes:*

* Updated Getters and Detail comment component.

* Updated filing template mixin to handle adding and parsing default detail comment.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/store/getters/state-getters.ts [hunk 1]
```diff
@@ -254,8 +254,9 @@ export const isFilingValid = (state: any): boolean => {
     state.stateModel.staffPaymentStep.valid)
 }
 
-export const getOriginalIAFilingDate = (state: any): boolean => {
-  return state.stateModel.originalIA.header.date.split('T')[0]
+export const getDefaultCorrectionDetailComment = (state: any): string => {
+  const filingDate = state.stateModel.originalIA.header.date.split('T')[0]
```
Comment (severinbeauvais) [inline src/store/getters/state-getters.ts:258]: Should you have a fallback value if original IA has not been loaded yet? (I know the object exists in the state model but there date=''.)

File: src/mixins/filing-template-mixin.ts [hunk 2]
```diff
@@ -251,7 +252,9 @@ export default class FilingTemplateMixin extends Vue {
       certifiedBy: filing.header.certifiedBy
     })
 
-    this.setDetailComment(filing.correction.comment)
+    const comment: string = filing.correction.comment || ''
+    const detailComment = comment.split('\n').slice(1).join('\n')
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:256]: This is complex enough that I think it needs a comment.
Might as well add a comment to line 110 to say what you're doing there too :)

**Review by severinbeauvais — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/mixins/filing-template-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply src/mixins/filing-template-mixin.ts:256]: Absolutely, will do!

File: src/store/getters/state-getters.ts (hunk 1 above)
Comment (cameron-freshworks) [reply src/store/getters/state-getters.ts:258]: Great call!

**Pull request merged**